### PR TITLE
CI: fix `efficientnet` pipeline timeout and prevent future similar issues due to large image size

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

This PR:
- Should fix the timeout we've been seeing in the efficientnet pipeline
- Adds exceptions to prevent similar cases from appearing again

### CODE REVIEW

File: tests/utils/tiny_model_summary.json [hunk 1]
```diff
@@ -1718,7 +1718,7 @@
         "model_classes": [
             "EfficientNetForImageClassification"
         ],
-        "sha": "6ed195ee636d2c0b885139da8c7b45d57ebaeee0"
+        "sha": "993d088cf937b8a90b61f68677cd8f261321c745"
```
Comment (gante) [inline tests/utils/tiny_model_summary.json:1721]: Corresponds to [this commit](https://huggingface.co/hf-internal-testing/tiny-random-EfficientNetForImageClassification/commit/993d088cf937b8a90b61f68677cd8f261321c745).

This is the processor config that results from the corrected workflow.

File: utils/create_dummy_models.py [hunk 2]
```diff
@@ -504,6 +504,27 @@ def convert_feature_extractor(feature_extractor, tiny_config):
...
+    # can result in timeout issues.
+    if (
+        isinstance(feature_extractor, BaseImageProcessor)
+        and hasattr(feature_extractor, "size")
+        and isinstance(feature_extractor.size, dict)
+    ):
+        largest_image_size = max(feature_extractor.size["height"], feature_extractor.size["width"])
+        if largest_image_size > 64:
+            # hardcoded exceptions
+            models_with_large_image_size = ("deformable_detr", "flava", "grounding_dino", "mgp_str", "swiftformer")
```
Comment (gante) [inline utils/create_dummy_models.py:517]: manually checked with a manual search using "image_size=" as search word, on the model tester files.

File: utils/create_dummy_models.py [hunk 3]
```diff
@@ -1395,7 +1416,7 @@ def create_tiny_models(
         raise ValueError(f"This script should be run from the root of the clone of `transformers` {clone_path}")
 
     report_path = os.path.join(output_path, "reports")
-    os.makedirs(report_path)
+    os.makedirs(report_path, exist_ok=True)
```
Comment (gante) [inline utils/create_dummy_models.py:1419]: before: we had to remove the reports folder each time we ran the command.
now: it overwrites the contents of the reports folder when we run the command.

File: utils/get_test_info.py [hunk 4]
```diff
@@ -53,7 +53,15 @@ def get_module_path(test_file):
...
-    test_module = importlib.import_module(test_module_path)
+    try:
+        test_module = importlib.import_module(test_module_path)
+    except AttributeError as exc:
+        # e.g. if you have a `tests` folder in `site-packages`, created by another package, when trying to import
+        # `tests.models...`
+        raise ValueError(
+            f"Could not import module {test_module_path}. Confirm that you don't have a package with the same root "
+            "name installed or in your environment's `site-packages`."
+        ) from exc
```
Comment (gante) [inline utils/get_test_info.py:64]: `importlib.import_module(test_module_path)` tries to import a local module. In my particular setup, the local module (`"tests"`) conflicted with a package that I had installed. 

I took a while to figure out why it was crashing, so I decided to enhance the exception message 👀

File: utils/create_dummy_models.py [hunk 5]
```diff
@@ -504,6 +504,27 @@ def convert_feature_extractor(feature_extractor, tiny_config):
...
+    if (
+        isinstance(feature_extractor, BaseImageProcessor)
+        and hasattr(feature_extractor, "size")
+        and isinstance(feature_extractor.size, dict)
+    ):
+        largest_image_size = max(feature_extractor.size["height"], feature_extractor.size["width"])
+        if largest_image_size > 64:
+            # hardcoded exceptions
+            models_with_large_image_size = ("deformable_detr", "flava", "grounding_dino", "mgp_str", "swiftformer")
+            if any(model_name in feature_extractor.__class__.__name__ for model_name in models_with_large_image_size):
```
Comment (ydshieh) [inline utils/create_dummy_models.py:518]: `feature_extractor.__class__.__name__` is the class name of a feature extractor, which is unlikely to contain model type names like `deformable_detr` or `grounding_dino`. Also the upper/lower cases should be considered.

Comment (ydshieh) [reply utils/create_dummy_models.py:518]: (or just use the class names in `models_with_large_image_size`)

File: utils/get_test_info.py (hunk 4 above)
Comment (ydshieh) [reply utils/get_test_info.py:64]: yeah, it happened to me before!

File: utils/create_dummy_models.py (hunk 2 above)
Comment (ydshieh) [reply utils/create_dummy_models.py:517]: does this mean those model could run pipeline tests with larger image size, so we don't care?

**Review by ydshieh — APPROVED** [review]
Thank you for taking care of this.

File: utils/create_dummy_models.py (hunk 2 above)
Comment (gante) [reply utils/create_dummy_models.py:517]: @ydshieh yeah -- that is what is happening at the moment. 

For a new model, if we need to use a larger image size in the tiny model, we would have to add it here. In other words, it works as an intentional source of friction.

File: utils/create_dummy_models.py [hunk 6]
```diff
@@ -504,6 +504,27 @@ def convert_feature_extractor(feature_extractor, tiny_config):
...
         feature_extractor = feature_extractor.__class__(**kwargs)
 
+    # Sanity check: on tiny image feature extractors, a large image size results in slow CI -- up to the point where it
+    # can result in timeout issues.
+    if (
+        isinstance(feature_extractor, BaseImageProcessor)
+        and hasattr(feature_extractor, "size")
+        and isinstance(feature_extractor.size, dict)
+    ):
+        largest_image_size = max(feature_extractor.size["height"], feature_extractor.size["width"])
```
Comment (amyeroberts) [inline utils/create_dummy_models.py:514]: This assumes that the keys in the `size` dict are `["height", "width"]`, but they could also be e.g. `["longest_edge", "shortest_edge"]`
```suggestion
        largest_image_size = max(feature_extractor.size.values())
```

**Review by amyeroberts — APPROVED** [review]
Thanks for digging into this and making our testing suite safer! 

extra points for the always sunny meme ;)

**Pull request merged**